### PR TITLE
Resource management

### PR DESCRIPTION
**Pull request by FHaggs**

### CODE REVIEW

File: resources/policies/datasources.rego [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package tauth.datasources
+
+import rego.v1
+
+admin_resources := _filter_resource("ds-admin:")
```
Comment (martinduartemore) [inline resources/policies/datasources.rego:5]: Let's use `::` as a separator. Example: `ds::admin::`.

We could use a policy with a function `build_permission_name(name_list)` that automatically builds a string using the separator. Then, all policies that need to build permission names can import this function from that policy. Example: `build_permission_name(["ds", "admin", "read-ID"]) = ds::admin::read-ID`.

File: tauth/dependencies/__init__.py [hunk 2]
```diff
@@ -7,4 +7,4 @@
 def init_app(app: FastAPI, sets: Settings) -> None:
     database.init_app(sets)
     authentication.init_app(app)
-    authorization.init_app()
+    authorization.setup_engine()
```
Comment (martinduartemore) [inline tauth/dependencies/__init__.py:10]: Reflect this change in docs.

File: tauth/entities/routes.py [hunk 3]
```diff
@@ -177,3 +179,77 @@ async def remove_entity_role(
...
+            status_code=s.HTTP_404_NOT_FOUND,
+            detail="Permission not found.",
+        )
+    logger.debug(f"Permission found: {permission!r}.")
+    # 409 in case the role is already attached
+    for p in entity.permissions:
+        if p == permission.id:
+            raise HTTPException(
+                status_code=s.HTTP_409_CONFLICT,
+                detail=f"Role {permission.name!r} already attached to entity {entity.handle!r}.",
```
Comment (martinduartemore) [inline tauth/entities/routes.py:216]: Role -> Permission.

File: tauth/entities/routes.py [hunk 4]
```diff
@@ -177,3 +179,77 @@ async def remove_entity_role(
...
+        f"Removing permission {permission_id!r} from entity {entity_id!r}."
+    )
+    entity_coll = EntityDAO.collection(alias=Settings.get().REDBABY_ALIAS)
+    res = entity_coll.update_one(
+        {"_id": entity_id},
+        {"$pull": {"permissions": permission_id}},
+    )
+    logger.debug(f"Update result: {res!r}.")
+    return {
+        "msg": "Role removed from entity.",
```
Comment (martinduartemore) [inline tauth/entities/routes.py:252]: Role -> Permission.

File: tauth/resource_management/access/models.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    Having a ResourceAcess does not mean anything other than that.
+    If you have READ, WRITE or ADMIN is up to the REGO Policy
+    """
+
+    resource_id: PyObjectId
+    entity_ref: EntityRef
+
+    @classmethod
+    def collection_name(cls) -> str:
+        return "resource-access"
```
Comment (martinduartemore) [inline tauth/resource_management/access/models.py:23]: `resource-accesses`

File: tauth/resource_management/access/routes.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+            detail=m,
+        )
+
+    return GeneratedFields(**resource_access.bson())
+
+
+@router.get("/{access_id}", status_code=s.HTTP_200_OK)
+async def read_one(
+    access_id: PyObjectId,
+    infostar: Infostar = Depends(privileges.is_valid_user),
```
Comment (martinduartemore) [inline tauth/resource_management/access/routes.py:68]: Use `Annotated`.

File: tauth/resource_management/resources/schemas.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+                                    "id": "thread-id",
+                                    "metadata": {"alias": "osf"},
+                                }
+                            ],
+                        },
+                    },
+                },
+            ]
+        }
+    )
```
Comment (martinduartemore) [inline tauth/resource_management/resources/schemas.py:42]: Let's use FastAPI's `Example` class for this.

File: tauth/utils/fastapi_extension.py
Comment (martinduartemore) [inline tauth/utils/fastapi_extension.py:1]: Can we delete this file? :ninja:

File: tauth/utils/headers.py
Comment (martinduartemore) [inline tauth/utils/headers.py:1]: HNNNGGGGGG :biting_lip: :hot_face: :sweat_drops:

**Review by martinduartemore — CHANGES_REQUESTED** [review]

File: resources/policies/datasources.rego (hunk 1 above)
Comment (FHaggs) [reply resources/policies/datasources.rego]: done

File: resources/policies/datasources.rego
Comment (martinduartemore) [inline resources/policies/datasources.rego:1]: :goat:

File: resources/policies/utils.rego
Comment (martinduartemore) [inline resources/policies/utils.rego:1]: :ok_hand:

**Review by martinduartemore — APPROVED** [review]

**Pull request merged**